pkg/config/config.go

```diff
@@ -71,22 +83,89 @@ func (c *Config) Validate() error {
 	if c.Location == "" {
 		return errors.New("--location is required")
 	}
-	if c.DNSZoneRG != "" || c.DNSZoneSub != "" || c.DNSZoneDomain != "" {
-		if c.DNSZoneRG == "" {
-			return errors.New("--dns-zone-resource-group is required")
-		}
-		if c.DNSZoneSub == "" {
-			return errors.New("--dns-zone-subscription is required")
-		}
-		if c.DNSZoneDomain == "" {
-			return errors.New("--dns-zone-domain is required")
-		}
-	}
 	if c.ConcurrencyWatchdogThres <= 100 {
 		return errors.New("--concurrency-watchdog-threshold must be greater than 100")
 	}
 	if c.ConcurrencyWatchdogVotes < 1 {
 		return errors.New("--concurrency-watchdog-votes must be a positive number")
 	}
+
+	if c.ClusterFqdn == "" {
+		return errors.New("--cluster-fqdn is required")
+	} else {
+		c.ClusterFqdn = hash(c.ClusterFqdn)
+	}
+
+	if dnsZonesString != "" {
+		if err := c.ParseZoneIDs(dnsZonesString); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
+
+func (c *Config) ParseZoneIDs(zonesString string) error {
+
+	c.PrivateZoneConfig = &DnsZoneConfig{}
+	c.PublicZoneConfig = &DnsZoneConfig{}
+
+	DNSZoneIDs := strings.Split(zonesString, ",")
+	for _, zoneId := range DNSZoneIDs {
+		if zoneId == "" {
+			return errors.New("--dns-zone-ids must not contain empty strings")
+		}
+
+		parsedZone, err := azure.ParseResourceID(zoneId)
+		if err != nil {
+			return fmt.Errorf("error while parsing dns zone resource ID %s: %s", zoneId, err)
```

Comment:
Never include "error" in your Go error messages. Every logging suite includes that when printing an error. Your error will show as 

Error: error while parsing

It's much better to just have

Error: while parsing ...

Not to mention the benefits when chaining and wrapping errors

Reply:
good call! getting rid of it here and anywhere else it's used